packages/core/src/agent/Dispatcher.ts

```diff
@@ -91,17 +92,21 @@ class Dispatcher {
   }
 
   private getHandlerForType(messageType: string): Handler | undefined {
+    const incomingMessageType = parseMessageType(messageType)
+
     for (const handler of this.handlers) {
       for (const MessageClass of handler.supportedMessages) {
-        if (MessageClass.type === messageType) return handler
+        if (canHandleMessageType(MessageClass, incomingMessageType)) return handler
       }
     }
   }
 
   public getMessageClassForType(messageType: string): typeof AgentMessage | undefined {
+    const incomingMessageType = parseMessageType(messageType)
+
     for (const handler of this.handlers) {
       for (const MessageClass of handler.supportedMessages) {
-        if (MessageClass.type === messageType) return MessageClass
+        if (canHandleMessageType(MessageClass, incomingMessageType)) return MessageClass
       }
     }
   }
```

Comment:
Same here, could be a bit cleaner.

Reply:
Same as above

Reply:
return this.handlers().flatMap((handler) => handler.supportedMessages).find((MessageClass) => canHandleMessageType(MessageClass, incomingMessageType))

Reply:
If `flatMap` is not supported:

```ts
return Array.prototype.concat(...this.handlers.map((handler) => handler.supportedMessages).find((MessageClass) => canHandleMessageType(MessageClass, incomingMessageType))
```